server/belga/io/feed_parsers/belga_iptc7901.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+import re
+import logging
+
+from superdesk.errors import ParserError
+from superdesk.io.registry import register_feed_parser
+from superdesk.io.feed_parsers.dpa_iptc7901 import DPAIPTC7901FeedParser
+from superdesk.metadata.utils import generate_guid
+from superdesk.metadata.item import ITEM_TYPE, CONTENT_TYPE, GUID_TAG
+from superdesk.utc import utcnow
+
+logger = logging.getLogger(__name__)
+
+
+class BelgaIPTC7901FeedParser(DPAIPTC7901FeedParser):
+    """
+    Feed Parser which can parse if the feed is in IPTC 7901 format.
+    """
+
+    NAME = 'belgaiptc7901'
+
+    label = 'Belga IPTC 7901 Parser'
+
+    types = [
+        ('dpa', b'([a-zA-Z]*)([0-9]*) (.) ([A-Z]{1,3}) ([0-9]*) ([a-zA-Z0-9 ]*)', [' =\n \n']),
+        ('ats', b'(\x7f\x7f|\x7f)', [' = \r\n \r\n', ' = \r\n\r\n'])
+    ]
+    txt_type = None
+
+    def can_parse(self, file_path):
+        try:
+            BelgaIPTC7901FeedParser.txt_type = None
+            with open(file_path, 'rb') as f:
+                lines = [line for line in f]
+                for item in self.types:
+                    check_type = re.match(item[1], lines[0], flags=re.I)
+                    if check_type:
+                        BelgaIPTC7901FeedParser.txt_type = item
+                        return check_type
+
+        except Exception:
+            return False
+
+    def parse(self, file_path, provider=None):
+        if BelgaIPTC7901FeedParser.txt_type[0] == 'dpa':
+            item = self.parse_content_dpa(file_path, provider)
+            item = self.dpa_derive_dateline(item)
+            self.parse_header(item)
+            # Markup the text and set the content type
+            item['body_html'] = '<p>' + item['body_html'].replace('\r\n', ' ').replace('\n', '</p><p>') + '</p>'
+            item[ITEM_TYPE] = CONTENT_TYPE.TEXT
+        if BelgaIPTC7901FeedParser.txt_type[0] == 'ats':
+            item = self.parse_content_ats(file_path, provider)
+            item = self.dpa_derive_dateline(item)
+            self.parse_header(item)
+            # Markup the text and set the content type
+            item['body_html'] = '<p>' + item['body_html'].replace('\r\n', ' ').replace('\n', '</p><p>') + '</p>'
+            item[ITEM_TYPE] = CONTENT_TYPE.TEXT
+
+        return item
```

Comment:
Exception might be here if `item !=  'dpa' and item !=  'ats'`

Reply:
init variable item = {}